clang/lib/3C/ConstraintBuilder.cpp

```diff
@@ -68,6 +70,47 @@ void processRecordDecl(RecordDecl *Declaration, ProgramInfo &Info,
   }
 }
 
+void processVarDecl(VarDecl *VD, ProgramInfo &Info,
+                    ASTContext *Context, ConstraintResolver CB) {
+  if(lastRecord2 != nullptr) {
+    uint lastRecordLocation = lastRecord2->getBeginLoc().getRawEncoding();
+    uint BeginLoc = VD->getBeginLoc().getRawEncoding();
+    uint EndLoc = VD->getEndLoc().getRawEncoding();
+    auto VarTy = VD->getType();
+    bool IsInLineStruct =
+        lastRecordLocation >= BeginLoc && lastRecordLocation <= EndLoc
```

Comment:
I can research this later if you don't know off the top of your head: Does `isBeforeInTranslationUnit` fire if they are not overlapping. For instance:
``` 
struct blah {...}; 
struct blah *x; 
``` 
since I'm not looking for the fact that a RecordDecl precedes the VarDecl but that the RecordDecl coincides with the VarDecl. I do agree that checking for source locations is a bit hacky, but I worry using this method will give me a lot of false positives.

Reply:
It actauly looks like there's am even better method: [isPointWithin](https://clang.llvm.org/doxygen/classclang_1_1SourceManager.html#ae9cb89a5499999d0dd2ef0bf1389ad3f). It look like that can directly check if a source location fall between two other locations, which I think I exactly what you're doing.